sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -127,6 +127,27 @@ class KotlinSensor(
         sensorDuration?.stop()
     }
 
+    private fun reuseCPDTokens(inputFile: InputFile,sensorContext: SensorContext): Boolean {
+        if (!sensorContext.isCacheEnabled) {
+            return false
+        }
+        val cpdTokensCache = CPDTokensCache(sensorContext.previousCache(), sensorContext.nextCache())
+        val previousTokens: List<CPDToken>? = cpdTokensCache.loadFromPreviousAnalysis(inputFile)
+        return if (previousTokens != null) {
+            val tokensToReuse = sensorContext.newCpdTokens().onFile(inputFile)
+            previousTokens.forEach { tokensToReuse.addToken(it.range, it.text) }
+            tokensToReuse.save()
```

Comment:
Another nice place where we could use a scoping function, e.g. `.apply`. Instead of saving the `tokensToReuse` to a new val and using it later on, we can restrict the scope of where the `tokensToReuse` is available and at the same time be more explicit about where side-effects happen, which involve this API. Something like:
```
            sensorContext.newCpdTokens().onFile(inputFile).apply {
                previousTokens.forEach { addToken(it.range, it.text) }
                save()
            }
```